examples/shader/automatic_instancing.rs

```diff
@@ -1,51 +1,97 @@
 //! Shows that multiple instances of a cube are automatically instanced in one draw call
 //! Try running this example in a graphics profiler and all the cubes should be only a single draw call.
+//! Also demonstrates how to use `MeshTag` to use external data in a custom material.
 
-use bevy::prelude::*;
+use bevy::{
+    prelude::*,
+    reflect::TypePath,
+    render::{
+        mesh::MeshTag,
+        render_resource::{AsBindGroup, ShaderRef},
+    },
+};
+
+const SHADER_ASSET_PATH: &str = "shaders/automatic_instancing.wgsl";
 
 fn main() {
     App::new()
-        .add_plugins(DefaultPlugins)
+        .add_plugins((DefaultPlugins, MaterialPlugin::<CustomMaterial>::default()))
         .add_systems(Startup, setup)
+        .add_systems(Update, update)
         .run();
 }
 
-/// set up a simple 3D scene
+/// Set up our scene
```

Comment:
This should probably have quick comment explaining the end result people should expect to see when running the example. It's not super obvious otherwise why it starts by loading a texture.